R/markdown.R

```diff
@@ -70,6 +71,32 @@ mdxml_unsupported <- function(xml, tag, feature) {
   xml_text(xml)
 }
 
+mdxml_code <- function(xml, tag) {
+  code <- xml_text(xml)
+
+  # See escaping details at
+  # https://cran.rstudio.com/doc/manuals/r-devel/R-exts.html#Insertions
+  if (can_parse(code)) {
+    paste0("\\code{", gsub("%", "\\\\%", code), "}")
+  } else {
+    paste0("\\verb{", escape_verb(code), "}")
+  }
+}
+
+can_parse <- function(x) {
+  tryCatch({
+    parse_expr(x)
+    TRUE
+  }, error = function(x) FALSE)
+}
+
+escape_verb <- function(x) {
+  x <- gsub("\\", "\\\\", x, fixed = TRUE)
+  x <- gsub("%", "\\%", x, fixed = TRUE)
+  x <- gsub("{", "\\{", x, fixed = TRUE)
+  x <- gsub("}", "\\}", x, fixed = TRUE)
+  x
+}
```

Comment:
This seems to produce the correct output, almost always, which is amazing! The only glitch I found is that `\%` is shown as `\\%` in the man page, and the same with ```` ``` ````. But maybe the PR #904 will fix this?

Reply:
Yeah, that's the job of #904.